File: 003.reverse-complement/reverse-complement.py
import sys

n_map = {
    'A': 'T',
    'T': 'A',
    'G': 'C',
    'C': 'G',
}

with open(sys.argv[1]) as f:
    DNA = f.read().strip()

print(
    ''.join(
        n_map[nucleotide] for nucleotide in DNA[::-1]
    )
)

File: 003.reverse-complement/reverse-complement.clj
; Ran with lein exec

(def nmap {\A \T, \G \C})

(def nucleotide-map
  (merge 
    nmap 
    (clojure.set/map-invert nmap))) ; Entirely unnecessary

(defn get-sequence 
  "Get DNA sequence from a file"
  [name-of-file]
  (slurp name-of-file))

(defn reverse-complement
  "Find the reverse complement"
  [DNA]
  (clojure.string/join "" (map nucleotide-map (reverse (seq DNA))))) ; Beautiful!

(println (reverse-complement (get-sequence (second *command-line-args*))))
